src/Traits/AdapterHelper.php

```diff
@@ -12,25 +12,25 @@
 
 trait AdapterHelper
 {
+    protected PropertyGroupOptionEntity $propertyGroupOptionEntity;
+    protected PluginConfig $pluginConfig;
+
     /**
      * @return Attribute[]
      */
-    public static function getPropertyGroupOptionAttributes(
+    protected function getPropertyGroupOptionAttributes(
         PropertyGroupOptionCollection $collection,
-        PluginConfig $pluginConfig,
+        PluginConfig $pluginConfig
     ): array {
+        $this->pluginConfig = $pluginConfig;
```

Comment:
The plugin config is already set within the constructor of both adapters, so you can remove it from all the function arguments.